enum_methods.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |i|
+      yield(i)
+    end
```

Comment:
The method `my_each` can't use the original ruby `each` method. You need to iterate your enumerable using a simple loop.